Realm/Realm/Handles/SharedRealmHandle.cs

```diff
@@ -420,6 +425,15 @@ private static unsafe void HandleOpenRealmCallback(IntPtr taskCompletionSource,
             }
         }
 
+        [MonoPInvokeCallbackAttribute(typeof(NativeMethods.FreeGCHandleCallback))]
+        public static void FreeGCHandle(IntPtr handle)
+        {
+            if (handle != IntPtr.Zero)
+            {
+                GCHandle.FromIntPtr(handle).Free();
```

Comment:
Tied to my renaming suggestion - I'm not a big fan of this approach because it leaks the internal design of the managed state. Instead, what we could do is have `State` implement `IDisposable` and in the callback, we can cast the target to `State` and call the `Dispose` method.